chapters/sampling-randomization-power.tex

```diff
@@ -1,529 +1,786 @@
-There are two distinct random processes referred to here:
-the conceptual process of assigning units to treatment arms,
-and the technical process of assigning random numbers in statistical software,
-which is a part of all tasks that include a random component.\sidenote{
-  \url{https://blog.stata.com/2016/03/10/how-to-generate-random-numbers-in-stata}}
-
-Any process that includes a random component is a random process,
-including sampling, randomization, power calculation simulations, and algorithms like bootstrapping.
-Randomization is challenging and its mechanics are unintuitive for the human brain.
-``True'' randomization is also nearly impossible to achieve for computers,
-which are inherently deterministic.\sidenote{
-  \url{https://www.random.org/randomness}}
-
-\subsection{Implementing random processes reproducibly in Stata}
-
-For statistical programming to be considered reproducible, it must be possible for the outputs of random processes
-can be re-obtained at a future time.\cite{orozco2018make}
-For our purposes, we will focus on what you need to understand
-in order to produce truly random results for your project using Stata,
-and how you can make sure you can get those exact results again in the future.
+\subsection{Programming reproducible random processes}
+
+% what it means for randomization to be reproducible
+For statistical programming to be reproducible,
+you must be able to re-obtain its exact outputs in the future.\cite{orozco2018make}
+We will focus on what you need to do to produce
+truly random results for your project,
+to ensure you can get those results again.
 This takes a combination of strict rules, solid understanding, and careful programming.
-This section introduces the strict rules: these are non-negotiable (but thankfully simple). 
-At the end of the section, 
-we provide a do-file that provides a concrete example of how to implement these principles. 
-
-Stata, like most statistical software, uses a \textbf{pseudo-random number generator}.
-Basically, it has a pre-calculated really long ordered list of numbers with the property that
-knowing the previous one gives you precisely zero information about the next one, i.e. a list of random numbers.
-Stata uses one number from this list every time it has a task that is non-deterministic.
-In ordinary use, it will cycle through these numbers starting from a fixed point
-every time you restart Stata, and by the time you get to any given script,
-the current state and the subsequent states will be as good as random.\sidenote{
-  \url{https://www.stata.com/manuals14/rsetseed.pdf}}
-However, for true reproducible randomization, we need two additional properties:
-we need to be able to fix the starting point so we can come back to it later;
-and we need to ensure that the starting point is independently random from our process.
-In Stata, this is accomplished through three command concepts:
+This section introduces the strict rules:
```

Comment:
```suggestion
This section introduces strict rules:
```

Reply:
it refers back to the "strict rules" mentioned in the previous sentence. Hence the "the"